src/adapters/net.cpp

```diff
@@ -130,6 +131,13 @@ namespace net {
     if (!*m_socketfd) {
       throw network_error("Failed to open socket");
     }
+    
+    m_status.mac = file_util::contents(NET_PATH + m_interface + "/address");
+    if (m_status.mac == "") {
+      throw system_error("No MAC address located");
```

Comment:
If the file reading fails somehow, we don't want to module to completely crash. Just, set the value to `NO_MAC` in that case.